sdk/src/main/java/io/opentelemetry/sdk/metrics/BaseInstrument.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics;
+
+import io.opentelemetry.metrics.Instrument;
+import io.opentelemetry.metrics.LabelSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class BaseInstrument<B> implements Instrument<B> {
```

Comment:
This class is not thread safe (and Instrument needs to be as you see the annotation). You either need to use a copy-on-write pattern for the boundCounters or a Concurrent map.

Reply:
oh definitely. This is 100% not production ready. I was mostly just trying to see if what I put together is a valid starting point.

Reply:
I removed all the state from this, because I think it needs to be held in the accumulator layer, rather than here.